File: app/src/main/java/istg/edu/ec/appEmprendeISTGDev/InicioActivity.java
package istg.edu.ec.appEmprendeISTGDev;

// Imports de tu código original
import android.Manifest;
import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.content.pm.PackageManager;
import android.content.res.Configuration;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.text.SpannableString;
import android.text.style.ForegroundColorSpan;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.view.WindowCompat; // Importante para diseño moderno
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.NavDestination;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.google.android.gms.auth.api.signin.GoogleSignIn;
import com.google.android.gms.auth.api.signin.GoogleSignInClient;
import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
import com.google.android.material.navigation.NavigationView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

// Imports de la librería In-App Update
import com.google.android.play.core.appupdate.AppUpdateManager;
import com.google.android.play.core.appupdate.AppUpdateManagerFactory;
import com.google.android.play.core.install.model.AppUpdateType;
import com.google.android.play.core.install.model.UpdateAvailability;
import com.google.android.play.core.appupdate.AppUpdateOptions;

import istg.edu.ec.appEmprendeISTGDev.data.model.UserModel;
import istg.edu.ec.appEmprendeISTGDev.databinding.ActivityInicioBinding;
import istg.edu.ec.appEmprendeISTGDev.viewModel.PerfilsViewModel;
import istg.edu.ec.appEmprendeISTGDev.viewModel.PermisosViewModel;
import istg.edu.ec.appEmprendeISTGDev.viewModel.UserViewModel;
import istg.edu.ec.appEmprendeISTGDev.data.model.PerfilModel;
import istg.edu.ec.appEmprendeISTGDev.utils.DeepLinkManager;

import static istg.edu.ec.appEmprendeISTGDev.utils.StatusBarUtilsKt.setStatusBarColor;

public class InicioActivity extends AppCompatActivity {

    private AppBarConfiguration mAppBarConfiguration;
    private ActivityInicioBinding binding;
    public FirebaseAuth mAuth;
    private GoogleSignInClient mGoogleSignInClient;
    private ProgressDialog progressDialog;
    private UserViewModel userViewModel;
    private PermisosViewModel permisosViewModel;
    private PerfilsViewModel perfilviewModel;
    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;
    private NavigationView navigationView;
    private NavController navController;

    // --- VARIABLES PARA IN-APP UPDATE ---
    private AppUpdateManager appUpdateManager;
    private static final int MY_REQUEST_CODE = 100;

    // --- CÓDIGO NUEVO: Lanzador de permisos (Tu amiga no tiene esto) ---
    private final ActivityResultLauncher<String> requestPermissionLauncher =
            registerForActivityResult(new ActivityResultContracts.RequestPermission(), isGranted -> {
                if (isGranted) {
                    Log.d("InicioActivity", "Permiso de notificaciones concedido");
                } else {
                    Log.d("InicioActivity", "Permiso de notificaciones denegado");
                }
            });

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Ajuste para pantallas edge-to-edge
        WindowCompat.setDecorFitsSystemWindows(getWindow(), true);

        // Barra azul, iconos blancos (false)
        setStatusBarColor(this, R.color.azul_navbar, false);

        // --- VERIFICACIÓN DE ACTUALIZACIÓN ---
        checkForAppUpdate();

        // --- PERMISOS (Vital para Android 13+) ---
        askNotificationPermission();

        // --- Inyección de vistas y ViewModels ---
        sharedPreferences = getSharedPreferences("MisPreferencias", Context.MODE_PRIVATE);
        editor = sharedPreferences.edit();
        userViewModel = new ViewModelProvider(this).get(UserViewModel.class);
        permisosViewModel = new ViewModelProvider(this).get(PermisosViewModel.class);
        perfilviewModel= new ViewModelProvider(this).get(PerfilsViewModel.class);
        binding = ActivityInicioBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        // --- Firebase Auth ---
        mAuth = FirebaseAuth.getInstance();
        FirebaseUser currentUser = mAuth.getCurrentUser();

        // --- NavigationView ---
        navigationView = binding.navView;

        if (currentUser != null) {
            String uid = currentUser.getUid();
            editor.putString("uid", uid).apply();

            permisosViewModel.checkAdminStatus(uid);
            permisosViewModel.getEsAdmin().observe(this, esAdmin -> {
                Menu menu = navigationView.getMenu();
                menu.findItem(R.id.opcionesAdministradorFragment)
                        .setVisible(esAdmin != null && esAdmin);
            });

            View headerView = navigationView.getHeaderView(0);
            TextView tvEmail = headerView.findViewById(R.id.textView);
            tvEmail.setText(currentUser.getEmail());

            userViewModel.consultarUsuario(currentUser.getEmail());
        } else {
            Log.e("InicioActivity", "Usuario no autenticado");
            startActivity(new Intent(this, MainActivity.class));
            finish();
            return;
        }

        // --- Toolbar & Drawer ---
        setSupportActionBar(binding.appBarInicio.toolbar);
        DrawerLayout drawer = binding.drawerLayout;
        mAppBarConfiguration = new AppBarConfiguration.Builder(
                R.id.perfilFragment,
                R.id.nav_home,
                R.id.misNegociosFragment,
                R.id.filtroBusquedaFragment,
                R.id.opcionesAdministradorFragment
        )
                .setOpenableLayout(drawer)
                .build();

        navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_inicio);
        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);

        drawer.addDrawerListener(new DrawerLayout.SimpleDrawerListener() {
            @Override
            public void onDrawerOpened(View drawerView) {
                NavDestination dest = navController.getCurrentDestination();
                int idToCheck = (dest != null) ? dest.getId() : R.id.nav_home;
                navigationView.setCheckedItem(idToCheck);
            }
        });

        // --- Google Sign-In ---
        GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                .requestIdToken(getString(R.string.default_web_client_id))
                .requestEmail()
                .build();
        mGoogleSignInClient = GoogleSignIn.getClient(this, gso);

        // --- ProgressDialog ---
        progressDialog = new ProgressDialog(this);
        progressDialog.setMessage("Cerrando sesión...");
        progressDialog.setCancelable(false);

        // --- Observador de Usuario Firestore ---
        userViewModel.getUsuario().observe(this, usuario -> {
            if (usuario == null) {
                guardarUsuario(currentUser);
            } else {
                editor.putString("uid", usuario.getUid()).apply();
            }
        });

        // --- Navigation Logic ---
        navigationView.setNavigationItemSelectedListener(menuItem -> {
            boolean handled = NavigationUI.onNavDestinationSelected(menuItem, navController);

            if (menuItem.getItemId() == R.id.filtroBusquedaFragment) {
                navController.navigate(R.id.filtroBusquedaFragment);
                handled = true;
            }
            if (menuItem.getItemId() == R.id.nav_share) {
                compartirApp();
                handled = true;
            }
            if (handled) {
                drawer.closeDrawers();
            }
            return handled;
        });

        // --- MANEJO DE DEEP LINK ---
        binding.getRoot().post(() -> {
            DeepLinkManager.INSTANCE.handleIncomingIntent(
                    this,
                    getIntent(),
                    R.id.nav_host_fragment_content_inicio,
                    R.id.revisarPublicacionesFragment
            );
        });
    }

    // --- DEEP LINKS CUANDO LA APP YA ESTÁ ABIERTA ---
    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        setIntent(intent);
        binding.getRoot().post(() -> {
            DeepLinkManager.INSTANCE.handleIncomingIntent(
                    this,
                    getIntent(),
                    R.id.nav_host_fragment_content_inicio,
                    R.id.revisarPublicacionesFragment
            );
        });
    }

    // --- VERIFICAR ACTUALIZACIONES ---
    private void checkForAppUpdate() {
        appUpdateManager = AppUpdateManagerFactory.create(getApplicationContext());
        appUpdateManager.getAppUpdateInfo().addOnSuccessListener(appUpdateInfo -> {
            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
                    && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.IMMEDIATE)) {
                try {
                    appUpdateManager.startUpdateFlowForResult(
                            appUpdateInfo,
                            this,
                            AppUpdateOptions.newBuilder(AppUpdateType.IMMEDIATE).build(),
                            MY_REQUEST_CODE
                    );
                } catch (Exception e) {
                    Log.e("InicioActivity", "Error al iniciar actualización", e);
                }
            }
        });
    }

    private void guardarUsuario(FirebaseUser fireuser) {
        if (fireuser == null) return;
        UserModel user = new UserModel(fireuser.getEmail(), fireuser.getDisplayName(), fireuser.getUid());
        userViewModel.saveUsuario(user);
        String uid = fireuser.getUid();
        PerfilModel objperfil = new PerfilModel();
        objperfil.setUid(uid);
        objperfil.setNombre(fireuser.getDisplayName());
        objperfil.setEmail(fireuser.getEmail());
        objperfil.setFoto(fireuser.getPhotoUrl().toString());
        perfilviewModel.savePerfil(objperfil);
    }

    private void signOut() {
        mAuth.signOut();
        mGoogleSignInClient.signOut().addOnCompleteListener(task -> {
            sharedPreferences.edit().clear().apply();
            startActivity(new Intent(this, MainActivity.class));
            finish();
        });
    }

    private void compartirApp() {
        Intent shareIntent = new Intent(Intent.ACTION_SEND);
        shareIntent.setType("text/plain");
        shareIntent.putExtra(Intent.EXTRA_TEXT,
                "👋 Hola, te invito a descargar *App Emprende ISTG* 🚀 \n" +
                        "👉 Descárgala aquí: https://play.google.com/store/apps/details?id=istg.edu.ec.appEmprendeISTGDev"
        );
        startActivity(Intent.createChooser(shareIntent, "Compartir vía"));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.inicio, menu);
        MenuItem item = menu.findItem(R.id.action_settings);
        SpannableString spanString = new SpannableString(item.getTitle());
        int nightModeFlags = getResources().getConfiguration().uiMode & Configuration.UI_MODE_NIGHT_MASK;
        boolean isNight = nightModeFlags == Configuration.UI_MODE_NIGHT_YES;
        int color = isNight ? Color.WHITE : Color.BLACK;
        spanString.setSpan(new ForegroundColorSpan(color), 0, spanString.length(), 0);
        item.setTitle(spanString);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_settings) {
            new AlertDialog.Builder(this)
                    .setTitle("Cerrar sesión")
                    .setMessage("¿Estás seguro?")
                    .setPositiveButton("Sí", (dialog, which) -> {
                        progressDialog.show();
                        signOut();
                    })
                    .setNegativeButton("No", (dialog, which) -> dialog.dismiss())
                    .show();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onSupportNavigateUp() {
        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
                || super.onSupportNavigateUp();
    }

    // --- LÓGICA DE PERMISOS (Tu código gana aquí) ---
    private void askNotificationPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
            if (checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS) == PackageManager.PERMISSION_GRANTED) {
                // Permiso ya concedido
            } else if (shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS)) {
                requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS);
            } else {
                requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS);
            }
        }
    }
}

File: app/src/main/java/istg/edu/ec/appEmprendeISTGDev/InicioActivity.kt
package istg.edu.ec.appEmprendeISTGDev

import android.Manifest
import android.app.AlertDialog
import android.app.ProgressDialog
import android.content.DialogInterface
import android.content.Intent
import android.content.SharedPreferences
import android.content.pm.PackageManager
import android.content.res.Configuration
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.text.SpannableString
import android.text.style.ForegroundColorSpan
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.WindowCompat
import androidx.drawerlayout.widget.DrawerLayout
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.google.android.gms.auth.api.signin.GoogleSignIn
import com.google.android.gms.auth.api.signin.GoogleSignInClient
import com.google.android.gms.auth.api.signin.GoogleSignInOptions
import com.google.android.gms.tasks.Task
import com.google.android.material.navigation.NavigationView
import com.google.android.play.core.appupdate.AppUpdateInfo
import com.google.android.play.core.appupdate.AppUpdateManager
import com.google.android.play.core.appupdate.AppUpdateManagerFactory
import com.google.android.play.core.appupdate.AppUpdateOptions
import com.google.android.play.core.install.model.AppUpdateType
import com.google.android.play.core.install.model.UpdateAvailability
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import istg.edu.ec.appEmprendeISTGDev.data.model.PerfilModel
import istg.edu.ec.appEmprendeISTGDev.data.model.UserModel
import istg.edu.ec.appEmprendeISTGDev.databinding.ActivityInicioBinding
import istg.edu.ec.appEmprendeISTGDev.utils.DeepLinkManager.handleIncomingIntent
import istg.edu.ec.appEmprendeISTGDev.utils.setStatusBarColor
import istg.edu.ec.appEmprendeISTGDev.viewModel.PerfilsViewModel
import istg.edu.ec.appEmprendeISTGDev.viewModel.PermisosViewModel
import istg.edu.ec.appEmprendeISTGDev.viewModel.UserViewModel

class InicioActivity : AppCompatActivity() {

    private lateinit var binding: ActivityInicioBinding
    private lateinit var mAuth: FirebaseAuth
    private lateinit var mGoogleSignInClient: GoogleSignInClient
    private lateinit var progressDialog: ProgressDialog
    private lateinit var sharedPreferences: SharedPreferences

    private lateinit var userViewModel: UserViewModel
    private lateinit var permisosViewModel: PermisosViewModel
    private lateinit var perfilViewModel: PerfilsViewModel

    private lateinit var navController: NavController
    private lateinit var mAppBarConfiguration: AppBarConfiguration

    private var appUpdateManager: AppUpdateManager? = null

    // ✔ CORRECCIÓN: firma correcta en Kotlin (solo 1 parámetro)
    private val requestPermissionLauncher = registerForActivityResult(
        ActivityResultContracts.RequestPermission()
    ) { isGranted: Boolean ->
        Log.d("InicioActivity", if (isGranted)
            "Permiso de notificaciones concedido"
        else
            "Permiso de notificaciones denegado"
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        WindowCompat.setDecorFitsSystemWindows(window, true)
        this.setStatusBarColor(R.color.azul_navbar, false)

        binding = ActivityInicioBinding.inflate(layoutInflater)
        setContentView(binding.root)

        checkForAppUpdate()
        askNotificationPermission()

        sharedPreferences = getSharedPreferences("MisPreferencias", MODE_PRIVATE)

        userViewModel = ViewModelProvider(this)[UserViewModel::class.java]
        permisosViewModel = ViewModelProvider(this)[PermisosViewModel::class.java]
        perfilViewModel = ViewModelProvider(this)[PerfilsViewModel::class.java]

        mAuth = FirebaseAuth.getInstance()
        val currentUser = mAuth.currentUser

        if (currentUser == null) {
            startActivity(Intent(this, MainActivity::class.java))
            finish()
            return
        }

        setSupportActionBar(binding.appBarInicio.toolbar)

        // ✔ Obtener NavController correctamente en Kotlin
        val navHostFragment =
            supportFragmentManager.findFragmentById(R.id.nav_host_fragment_content_inicio) as NavHostFragment
        navController = navHostFragment.navController

        mAppBarConfiguration = AppBarConfiguration(
            setOf(
                R.id.perfilFragment,
                R.id.nav_home,
                R.id.misNegociosFragment,
                R.id.filtroBusquedaFragment,
                R.id.opcionesAdministradorFragment
            ),
            binding.drawerLayout
        )

        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration)

        // ✔ Firma correcta — ahora sí override válido
        binding.drawerLayout.addDrawerListener(object : DrawerLayout.SimpleDrawerListener() {
            override fun onDrawerOpened(drawerView: View) {
                navigationViewChecked()
            }
        })

        setupGoogleSignIn()
        setupUserData(currentUser)
        setupNavigationView()

        // Deep link
        binding.root.post {
            handleIncomingIntent(
                this,
                intent,
                R.id.nav_host_fragment_content_inicio,
                R.id.revisarPublicacionesFragment
            )
        }
    }

    override fun onNewIntent(intent: Intent?) {
        super.onNewIntent(intent)
        setIntent(intent)
        binding.root.post {
            handleIncomingIntent(
                this,
                getIntent(),
                R.id.nav_host_fragment_content_inicio,
                R.id.revisarPublicacionesFragment
            )
        }
    }

    private fun navigationViewChecked() {
        val dest = navController.currentDestination?.id ?: R.id.nav_home
        binding.navView.setCheckedItem(dest)
    }

    // ================= GOOGLE SIGN IN =================

    private fun setupGoogleSignIn() {
        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
            .requestIdToken(getString(R.string.default_web_client_id))
            .requestEmail()
            .build()

        mGoogleSignInClient = GoogleSignIn.getClient(this, gso)

        progressDialog = ProgressDialog(this).apply {
            setMessage("Cerrando sesión...")
            setCancelable(false)
        }
    }

    // ================= USUARIO FIRESTORE =================

    private fun setupUserData(user: FirebaseUser) {
        val uid = user.uid
        sharedPreferences.edit().putString("uid", uid).apply()

        permisosViewModel.checkAdminStatus(uid)
        permisosViewModel.esAdmin.observe(this) { esAdmin ->
            binding.navView.menu.findItem(R.id.opcionesAdministradorFragment)
                .isVisible = esAdmin == true
        }

        val header = binding.navView.getHeaderView(0)
        header.findViewById<TextView>(R.id.textView).text = user.email

        userViewModel.consultarUsuario(user.email!!)
        userViewModel.usuario.observe(this) { usuario ->
            if (usuario == null) {
                guardarUsuario(user)
            }
        }
    }

    private fun guardarUsuario(fireuser: FirebaseUser) {
        val user = UserModel(
            fireuser.email!!,
            fireuser.displayName ?: "",
            fireuser.uid
        )
        userViewModel.saveUsuario(user)

        val perfil = PerfilModel().apply {
            uid = fireuser.uid
            nombre = fireuser.displayName ?: ""
            email = fireuser.email!!
            foto = fireuser.photoUrl?.toString() ?: ""
        }
        perfilViewModel.savePerfil(perfil)
    }

    // ================= NAVIGATION =================

    private fun setupNavigationView() {
        binding.navView.setNavigationItemSelectedListener { item ->

            var handled = NavigationUI.onNavDestinationSelected(item, navController)

            if (item.itemId == R.id.nav_share) {
                compartirApp()
                handled = true
            }

            binding.drawerLayout.closeDrawers()
            handled
        }
    }

    // ================= PERMISOS =================

    private fun askNotificationPermission() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {

            when {
                checkSelfPermission(Manifest.permission.POST_NOTIFICATIONS)
                        == PackageManager.PERMISSION_GRANTED -> {}

                shouldShowRequestPermissionRationale(Manifest.permission.POST_NOTIFICATIONS) -> {
                    requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
                }

                else -> requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS)
            }
        }
    }

    // ================= IN-APP UPDATE =================

    private fun checkForAppUpdate() {
        appUpdateManager = AppUpdateManagerFactory.create(this)
        appUpdateManager!!.appUpdateInfo
            .addOnSuccessListener { info: AppUpdateInfo ->
                if (
                    info.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE &&
                    info.isUpdateTypeAllowed(AppUpdateType.IMMEDIATE)
                ) {
                    appUpdateManager!!.startUpdateFlowForResult(
                        info,
                        this,
                        AppUpdateOptions.newBuilder(AppUpdateType.IMMEDIATE).build(),
                        MY_REQUEST_CODE
                    )
                }
            }
    }

    // ================= COMPARTIR APP =================

    private fun compartirApp() {
        val shareIntent = Intent(Intent.ACTION_SEND).apply {
            type = "text/plain"
            putExtra(
                Intent.EXTRA_TEXT,
                "👋 Hola, te invito a descargar *App Emprende ISTG* 🚀\n" +
                        "👉 https://play.google.com/store/apps/details?id=istg.edu.ec.appEmprendeISTGDev"
            )
        }
        startActivity(Intent.createChooser(shareIntent, "Compartir vía"))
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.inicio, menu)

        val item = menu.findItem(R.id.action_settings)
        val spanString = SpannableString(item.title)

        val isNight =
            (resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK) ==
                    Configuration.UI_MODE_NIGHT_YES

        spanString.setSpan(
            ForegroundColorSpan(if (isNight) Color.WHITE else Color.BLACK),
            0,
            spanString.length,
            0
        )

        item.title = spanString
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_settings) {
            AlertDialog.Builder(this)
                .setTitle("Cerrar sesión")
                .setMessage("¿Estás seguro?")
                .setPositiveButton("Sí") { _, _ ->
                    progressDialog.show()
                    signOut()
                }
                .setNegativeButton("No", null)
                .show()

            return true
        }
        return super.onOptionsItemSelected(item)
    }

    private fun signOut() {
        mAuth.signOut()
        mGoogleSignInClient.signOut().addOnCompleteListener {
            sharedPreferences.edit().clear().apply()
            startActivity(Intent(this, MainActivity::class.java))
            finish()
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
                || super.onSupportNavigateUp()
    }

    companion object {
        private const val MY_REQUEST_CODE = 100
    }
}
